qa/L0_backend_python/python_unittest.py

```diff
@@ -55,32 +55,14 @@ def _run_unittest(self, model_name):
 
     def test_python_unittest(self):
         model_name = os.environ["MODEL_NAME"]
-        bls_kind = os.environ.get("BLS_KIND", "non_decoupled")
 
-        if bls_kind == "decoupled":
-            # Skip the shared memory probe for decoupled models for now as
-            # there are some small changes in the shared memory usage when
-            # running decoupled inferences. Confirmed that the memory growth
-            # is bounded.
-            self._run_unittest(model_name)
-        else:
-            if (
-                model_name == "bls"
-                or model_name == "bls_memory"
-                or model_name == "bls_memory_async"
-            ):
-                # For these tests, the memory region size will be grown. Because of
-                # this we need to use the shared memory probe only on the later
-                # call so that the probe can detect the leak correctly.
-                self._run_unittest(model_name)
+        # Warmup model, let it do any upfront initializations required
+        self._run_unittest(model_name)
 
-                # [FIXME] See DLIS-3684
+        with self._shm_leak_detector.Probe() as shm_probe:
+            for i in range(5):
+                print(f"[{model_name}] Test iteration: {i}")
```

Comment:
@Tabrizian does this change make sense? 

I'm trying to express:
1. Do any upfront "warmup" that would take some initial amount of memory for every use case
2. Then probe several follow-up runs to make sure the memory doesn't increase over the initial required upper bound

Reply:
Just want to sanity check that logic makes sense, and that I'm not potentially missing a leak by doing it this way.

Reply:
I think it is probably fine from leak detection but would significantly affect the test time since now we would run each test 5 times.

Reply:
Yeah I'll play with the numbers, just wanted to confirm the methodology

Reply:
@rmccorm4 I'm trying to understand you implementation here on warmup. The [existing implementation](https://github.com/triton-inference-server/server/blob/main/qa/L0_backend_python/python_unittest.py#L60-L82) will do a warmup only if the model name matches `bls`, `bls_memory`, `bls_memory_async` or `bls_request_rescheduling`. A warmup is not performed for all other model names. On your implementation, it will perform a warmup for all model names. Am I understanding it correctly? Why do we want to perform a warmup for all models?

Reply:
I think my goal was just to standardize the test for every use case, rather than having special cases as much as possible. If it significantly impacts total test time, it's probably not a good idea. This is just something I was experimenting with at the time.

Reply:
If you have any similar thoughts but want to do it differently, feel free - I don't think my latest changes were passing CI when I was last looking into this.

Reply:
Interestingly, I got it passed locally, after merging in main changes. I will push them here and run the CI, and see what happens.